libraries/botframework-connector/src/auth/parameterizedBotFrameworkAuthentication.ts

```diff
@@ -0,0 +1,475 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { ConnectorFactory } from '../connectorFactory';
+import { BotFrameworkClient } from '../skills';
+import { AuthenticateRequestResult } from './authenticateRequestResult';
+import { AuthenticationConfiguration } from './authenticationConfiguration';
+import { AuthenticationError } from './authenticationError';
+import { BotFrameworkAuthentication } from './botFrameworkAuthentication';
+import { BotFrameworkClientImpl } from './botFrameworkClientImpl';
+import { Claim, ClaimsIdentity } from './claimsIdentity';
+import { JwtTokenValidation } from './jwtTokenValidation';
+import { ServiceClientCredentialsFactory } from './serviceClientCredentialsFactory';
+import { SkillValidation } from './skillValidation';
+import { Activity, Channels, RoleTypes, StatusCodes } from 'botframework-schema';
+import { AuthenticationConstants } from './authenticationConstants';
+import { GovernmentConstants } from './governmentConstants';
+import { JwtTokenExtractor } from './jwtTokenExtractor';
+import { VerifyOptions } from 'jsonwebtoken';
+import { EmulatorValidation } from './emulatorValidation';
+import { ConnectorClientOptions } from '../connectorApi/models';
+import { ConnectorFactoryImpl } from './connectorFactoryImpl';
+import { UserTokenClientImpl } from './userTokenClientImpl';
+import { UserTokenClient } from './userTokenClient';
+
+// Clean up tokenValidationParameters
+
+// Internal
+export class ParameterizedBotFrameworkAuthentication extends BotFrameworkAuthentication {
+    private static SkillAndEmulatorTokenValidationParameters: VerifyOptions = {
```

Comment:
This isn't intended to support overrides and while TypeScript would prevent subclasses from overriding this static member, it would be overridable in JS. I'll change it to a top-level `const` instead. Good catch!

Reply:
I created an internally exported `const ToBotFromBotOrEmulatorTokenValidationParameters: VerifyOptions` to minimize the duplication across the authentication code and updated the exported `EmulatorValidation.ToBotFromEmulatorTokenValidationParameters` to forward the same value.